draft-ietf-jsonpath-base.md

```diff
@@ -284,7 +286,10 @@ stable JSON query language.
 
 ## JSON Values
 
-The JSON value the JSONPath query is applied to is, by definition, a valid JSON value.
+The JSON value a JSONPath query is applied to is, by definition, a
+valid JSON value. A JSON value is usually constructed by parsing
+a JSON text.
```

Comment:
@cabo I'm not sure this says quite enough. It would be _possible_ to parse a JSON text into a structure in which strings included their surrounding double quotes. The problem is that RFC 8259 doesn't define an abstract JSON value. Values in that RFC are textual representations, as the following ABNF extract makes clear (e.g. values can be objects with left curly braces):
```
value = false / null / true / object / array / number / string
...
object = begin-object [ member *( value-separator member ) ]
               end-object
...
begin-object    = ws %x7B ws  ; { left curly bracket
```
I'm not sure where the notion of a JSON value as an abstract concept came from, but it sorely needs defining.

Reply:
An object doesn't have curly braces.  Its representation in a JSON text does.

> the notion of a JSON value as an abstract concept came from, but it sorely needs defining.

Absolutely.  But we can't fix RFC 8259 here.  (Well, maybe we can, by simply providing the generic data model that JSON didn't define, but that might be a distraction.)

Reply:
> Also, RFC 8259 positions a value as nothing more than a JSON text minus any external whitespace:
> 
> ```
> JSON-text = ws value ws
> ```

Yes.  Each ABNF rule name should be called representation-for-x, representation-for-false, representation-for-object, representation-for-string.  This would get tedious quickly, so they left it off, leaving it for the reader to figure out that the ABNF defines the representation in a JSON text.

Reply:
This may go down like a lead balloon, but I've come clean about our assumption of an abstract JSON value without going so far as to define it.